marimo/_runtime/dataflow.py

```diff
@@ -158,6 +158,14 @@ def register_cell(self, cell_id: CellId_t, cell: CellImpl) -> None:
                         self.cycles.add(tuple([(other_id, cell_id)] + path))
                     self.children[other_id].add(cell_id)
 
+        if any(self.cells[cid].stale for cid in self.ancestors(cell_id)):
```

Comment:
this would be a good tiny function on the class `self._is_any_ancestor_slate(cell_id)`